src/types/blob.ts

```diff
@@ -1,30 +1,35 @@
-import { makeClass } from "../utils/make-class";
-import { parseBytes } from "../utils/bytes-utils";
-import { SerializedType } from "./serialized-type";
+import { SerializedTypeClass } from "./serialized-type";
+import { BinaryParser } from "../serdes/binary-parser";
 
-const Blob = makeClass(
-  {
-    mixins: SerializedType,
-    Blob(bytes) {
-      if (bytes) {
-        this._bytes = parseBytes(bytes, Uint8Array);
-      } else {
-        this._bytes = new Uint8Array(0);
-      }
-    },
-    statics: {
-      fromParser(parser, hint) {
-        return new this(parser.read(hint));
-      },
-      from(value) {
-        if (value instanceof this) {
-          return value;
-        }
-        return new this(value);
-      },
-    },
-  },
-  undefined
-);
+/**
+ * Variable length encoded type
+ */
+class Blob extends SerializedTypeClass {
+  constructor(bytes: Buffer) {
+    super();
+    this.bytes = bytes ?? Buffer.alloc(0);
```

Comment:
Where is `bytes` defined? If it's in the super class, maybe we should move this logic there?

Reply:
Yeah, bytes is defined in the super class, and is defined as `protected bytes: Buffer = Buffer.alloc(0);`, so it should already default to that.

If i change this to `if(bytes) {this.bytes = bytes}` it would work the same as `this.bytes = bytes ?? Buffer.alloc(0)`